src/lib/autoinstall/xml_checks.rb

```diff
@@ -31,35 +32,46 @@ module Y2Autoinstallation
   # for not well formed or invalid documents. It is possible to continue
   # anyway and ignore the found problems at your risk.
   class XmlChecks
-    extend Yast::Logger
-    extend Yast::I18n
-    textdomain "autoinst"
+    include Singleton
+    include Yast::Logger
+    include Yast::I18n
 
     # paths to the default AutoYaST schema files
     PROFILE_SCHEMA = "/usr/share/YaST2/schema/autoyast/rng/profile.rng".freeze
     RULES_SCHEMA = "/usr/share/YaST2/schema/autoyast/rng/rules.rng".freeze
     CLASSES_SCHEMA = "/usr/share/YaST2/schema/autoyast/rng/classes-use.rng".freeze
 
+    ERRORS_PATH = "/var/lib/YaST2/xml_checks_errors".freeze
+    # @return [Array<String>] reported errors md5
+    attr_reader :reported_errors
+
+    # Constructor
+    def initialize
+      textdomain "autoinst"
+
+      @reported_errors = errors_from_file
+    end
+
     # Is the file a valid AutoYaST XML profile?
     # @param file [String] pahth to the classes file
     # @return [Boolean] true if valid, false otherwise
-    def self.valid_profile?(file = Yast::AutoinstConfig.xml_tmpfile)
+    def valid_profile?(file = Yast::AutoinstConfig.xml_tmpfile)
       # TRANSLATORS: Error message
       msg = _("The AutoYaST profile is not a valid XML document.")
       check(file, PROFILE_SCHEMA, msg)
     end
 
     # Is the file a valid AutoYaST XML profile?
     # @return [Boolean] true if valid, false otherwise
-    def self.valid_modified_profile?
+    def valid_modified_profile?
       # TRANSLATORS: Error message
       msg = _("The AutoYaST pre-script generated an invalid XML document.")
```

Comment:
maybe generate instead of generated?

Reply:
For me looks ok, and it is not adde by me but already there.